Add iterator to pini

NikolaJelic opened the pull request:
Added:
- iterator to the `pini` class that uses the iterator built into `std::unordered_map`
- wrapper functions for `std::unordered_map`'s `empty()`, `size()` `clear()` and `erase()`

NikolaJelic committed:
add iterator to the pini object

karnkaul commented on include/pini/pini.hpp:
```diff
@@ -16,6 +17,13 @@ class pini {
 
+	// iteration
+
+	using const_iterator = std::unordered_map<std::string, std::string>::const_iterator;
```
Just curious: any reason you don't want to alias the map type itself, and instead write down that long type name every time?

karnkaul replied:
I see that you do have the alias as `map_type`, so why not use that here?!

karnkaul commented on include/pini/pini.hpp:
```diff
@@ -35,6 +43,13 @@ class pini {
 	static void default_callback(std::string_view msg, severity level);
 	inline static on_msg on_msg_t = &default_callback;
 
+	bool is_empty();
```
Missing const correctness

karnkaul commented on include/pini/pini.hpp:
```diff
@@ -35,6 +43,13 @@ class pini {
 	inline static on_msg on_msg_t = &default_callback;
 
+	bool is_empty();
+	std::size_t size();
```
Missing const correctness

karnkaul commented on src/pini.cpp:
```diff
@@ -89,4 +89,11 @@ void pini::default_callback(std::string_view msg, severity level) {
 }
 
+bool pini::is_empty() { return key_value_pairs.empty(); }
+std::size_t pini::size() { return key_value_pairs.size(); }
```
You may want to inline `is_empty()` and `size()` (implement in header / in class definition)

karnkaul commented on test/pini_test.cpp:
```diff
@@ -5,10 +5,11 @@
+	if (!pin.load_file(filename)) { return 1; }
+	for (auto& pair : pin) { std::cout << pair.first << " " << pair.second << '\n'; }
+	std::cout << pin.size() << " | " << pin.is_empty();
+	pin.erase("def");
```
This needs more failure cases. Like I said before, you aren't going to see any output in automated runs.

NikolaJelic committed:
update PR
- moved util to pn namespace,
- used callbacks instead of cerr for data converters,
- inlined size() and empty()

NikolaJelic committed:
imptoved testing
